hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/TestProtoKafkaSource.java

```diff
@@ -198,22 +235,48 @@ private static Nested generateRandomNestedMessage() {
   @Override
   void sendMessagesToKafka(String topic, int count, int numPartitions) {
     List<Sample> messages = createSampleMessages(count);
-    try (Producer<String, byte[]> producer = new KafkaProducer<>(getProducerProperties())) {
+    try (Producer<String, byte[]> producer = new KafkaProducer<>(getProducerProperties(false))) {
       for (int i = 0; i < messages.size(); i++) {
         // use consistent keys to get even spread over partitions for test expectations
         producer.send(new ProducerRecord<>(topic, Integer.toString(i % numPartitions), messages.get(i).toByteArray()));
       }
     }
   }
 
-  private Properties getProducerProperties() {
+  private void sendMessagesToKafkaWithConfluentSerializer(String topic, int numPartitions, List<Sample> messages) {
+    try (Producer<String, Message> producer = new KafkaProducer<>(getProducerProperties(true))) {
+      for (int i = 0; i < messages.size(); i++) {
+        // use consistent keys to get even spread over partitions for test expectations
+        producer.send(new ProducerRecord<>(topic, Integer.toString(i % numPartitions), messages.get(i)));
+      }
+    }
+  }
+
+  private Properties getProducerProperties(boolean useConfluentProtobufSerializer) {
     Properties props = new Properties();
     props.put("bootstrap.servers", testUtils.brokerAddress());
-    props.put("value.serializer", ByteArraySerializer.class.getName());
-    // Key serializer is required.
+    if (useConfluentProtobufSerializer) {
+      props.put("value.serializer",
+          "io.confluent.kafka.serializers.protobuf.KafkaProtobufSerializer");
+      props.put("value.deserializer",
+          "io.confluent.kafka.serializers.protobuf.KafkaProtobufDeserializer");
```

Comment:
Instead of hard-coding, how about `KafkaProtobufSerializer.class.getName()`? Is it that we don't want to load the imports of external class?

Reply:
At the time, my intellij imports were not working. Updating this now